src/Pages/Withdrawals.razor

```diff
@@ -68,15 +68,16 @@
                     <DataGridColumn TItem="WalletWithdrawalRequest" Filterable="false" Caption="" Sortable="false" Displayable="true">
                         <DisplayTemplate>
                             @*TODO Cancel / Reject  *@
-                            @if ((context.Status == WalletWithdrawalRequestStatus.Pending || context.Status == WalletWithdrawalRequestStatus.PSBTSignaturesPending) && LoggedUser?.Id == context.UserRequestorId)
-
-                            {
-                                <Button Color="Color.Danger" Clicked="()=> ShowRejectCancelModal(context, WalletWithdrawalRequestStatus.Cancelled)">Cancel</Button>
+                            @{
+                                bool canCancel = context.Status == WalletWithdrawalRequestStatus.Pending || context.Status == WalletWithdrawalRequestStatus.PSBTSignaturesPending;
+                                bool isRequestor = LoggedUser?.Id == context.UserRequestorId;
+                                string action = isRequestor ? "Cancel" : "Reject";
+                                WalletWithdrawalRequestStatus newStatus = isRequestor ? WalletWithdrawalRequestStatus.Cancelled : WalletWithdrawalRequestStatus.Rejected;
                             }
-                            else if ((context.Status == WalletWithdrawalRequestStatus.Pending || context.Status == WalletWithdrawalRequestStatus.PSBTSignaturesPending)
-                                     && LoggedUser?.Id != context.UserRequestorId)
+                            
+                            @if (canCancel || (canCancel && !isRequestor))
```

Comment:
This codes works like this:
if canCancel is true, the first part is true, the code displays the button
if canCancel is false, first part is false, the second part is also false, the code doesn't display the button

so you only need the first part:
```
@if (canCancel) {
```

Reply:
Sure, I missed.